src/couch/src/couch_server.erl

```diff
@@ -465,11 +470,12 @@ handle_call({delete, DbName, Options}, _From, Server) ->
     Error ->
         {reply, Error, Server}
     end;
-handle_call({db_updated, #db{}=Db}, _From, Server0) ->
-    #db{name = DbName, instance_start_time = StartTime} = Db,
-    Server = try ets:lookup_element(couch_dbs, DbName, #db.instance_start_time) of
+handle_call({db_updated, Db}, _From, Server0) ->
+    DbName = couch_db:name(Db),
+    StartTime = couch_db:get_instance_start_time(Db),
+    Server = try ets:lookup_element(couch_dbs, DbName, #entry.start_time) of
         StartTime ->
-            true = ets:insert(couch_dbs, Db),
+            true = ets:update_element(couch_dbs, DbName, {#entry.db, Db}),
```

Comment:
`ets:update_element` will return false if it could not find the entry and it would crash on a bad match, while previously `ets:insert` always returned `true` and inserted the entry in the table (even if it wasn't in the table already).

Is there any change the entry could be deleted between the call to ets:lookup and ets:update_element?